sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala

```diff
@@ -34,7 +34,7 @@ import org.apache.spark.util.Utils
 // The data where the partitioning key exists only in the directory structure.
 case class ParquetData(intField: Int, stringField: String)
 // The data that also includes the partitioning key
-case class ParquetDataWithKey(p: Int, intField: Int, stringField: String)
+case class ParquetDataWithKey(pQ: Int, intField: Int, stringField: String)
```

Comment:
The modified unit tests did surface a bug, but it's tricky in how it manifested itself. Very briefly, when adding missing partitions for partition columns with an upper-case letter `msck repair table` works, but `alter table ... add partition` does not. The latter actually creates new directories. For example, here's a partitioned table listing before calling `add partition`:

```
-rw-r--r--   3 msa msa          0 2016-10-14 03:52 test_mixed_case/_SUCCESS
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=0
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=1
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=2
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=3
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=4
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=5
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=6
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=7
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=8
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=9
```

Here's the listing after calling `add partition` for each partition in turn:

```
-rw-r--r--   3 msa msa          0 2016-10-14 03:52 test_mixed_case/_SUCCESS
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=0
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=1
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=2
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=3
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=4
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=5
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=6
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=7
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=8
drwxr-xr-x   - msa msa          0 2016-10-14 03:52 test_mixed_case/partCol=9
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=0
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=1
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=2
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=3
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=4
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=5
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=6
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=7
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=8
drwxr-xr-x   - msa msa          0 2016-10-14 03:53 test_mixed_case/partcol=9
```

I'll file an issue when I have a moment.